sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs

```diff
@@ -61,6 +61,28 @@ public void Ctor_ConnectionString()
             Assert.AreEqual("accountName", builder2.AccountName);
         }
 
+        [Test]
+        [Ignore("Test framework doesn't allow recorded tests with connection string because the word 'Sanitized' is not base-64 encoded, so we can't pass connection string validation")]
+        public async Task Ctor_ConnectionStringEscapeBlobName()
+        {
+            // Arrange
+            await using DisposingContainer test = await GetTestContainerAsync();
+            string blobName = "!*'();[]:@&%=+$,/?#";
```

Comment:
As far as I recall (@rickle-msft plz chime in if I'm wrong here) @rickle-msft  was/is chasing a bug in Java where [German Umlauts](https://learn-german-easily.com/german-umlauts) got encoded with different target encoding (turns out Url encoding/percent encoding has target encoding...) than expected by server side. Might be worth adding that to test case suite.

Reply:
what's target encoding?

Reply:
https://www.urlencoder.org/

![image](https://user-images.githubusercontent.com/61715331/80645829-88630a00-8a20-11ea-9f3f-ac2d5a3ca7cf.png)

![image](https://user-images.githubusercontent.com/61715331/80645885-9b75da00-8a20-11ea-9197-9a9fc43f41a2.png)

Reply:
[Here's](https://github.com/Azure/azure-sdk-for-java/pull/10273) the PR where this was addressed in Java.

Reply:
Added German characters to both tests, they are still passing.